### DISCUSSION TITLE
Are state ids same everywhere?

### DISCUSSION THREAD

**Discussion by maxsupermanhd** [Tnze/go-mc#156] [Q&A] [answered]
I am trying to port color generator to new blockstates model and so far I have question: how to store it?

My idea is that I store a `map[stateid]color` in gob for example and load it each time I need it, will state ids be same for same versions?

**Comment by Tnze** [comment]
No, they aren't. And that's why the game save store the blocks' string id and properties. The integer id is only used in runtime and shouldn't be stored.

**Comment by Tnze** [comment, marked as answer]
oh, they're same for a particular version

**Comment by maxsupermanhd** [comment]
I just built a parser of blockstate jsons that ship inside mc jar, throw id and description at it and it will return a block.Block
